app/models/work.rb

```diff
@@ -411,7 +423,10 @@ def save_new_recipients
     unless self.new_recipients.blank?
       self.new_recipients.split(',').each do |name|
         gift = self.gifts.for_name_or_byline(name).first
-        self.gifts << Gift.new(:recipient => name) unless gift
+        unless gift
+          g = Gift.new(recipient: name, work: self)
+          g.save if g.valid?
```

Comment:
You never really need to check valid? manually before saving, since that will happen automatically. If it's not valid, .save will just return false. (Is there any other reason to do it that way?)